### PR TITLE
CA Driver Overhaul

### PR DESCRIPTION
**Pull request by wavious-wpatty**
Fixes:
  - None

Features:
  - CA Drivers overhaul

### CODE REVIEW

File: include/drivers/ca/ca_bscan.h [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+#include <stdbool.h>
+#include "ca_reg.h"
+
+/*******************************************************************************
+**                            CA
+*******************************************************************************/
+/**
+ * @brief   CA DQ BSCAN Get Status Register Interface
+ *
+ * @details Gets the CA slice BSCAN status.
```
Comment (wavious-jbasista) [inline include/drivers/ca/ca_bscan.h:18]: Is this supposed to say DQ slice BSCAN status?

File: include/drivers/ca/ca_bscan.h [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ */
+void ca_dq_bscan_get_status_reg_if(ca_reg_t *ca_reg, uint8_t *status);
+
+/*******************************************************************************
+**                            CK
+*******************************************************************************/
+/**
+ * @brief   CA DQS BSCAN Get Status Register Interface
+ *
+ * @details Gets the CK slice BSCAN status.
```
Comment (wavious-jbasista) [inline include/drivers/ca/ca_bscan.h:33]: Is this supposed to say DQS slice BSCAN status?

File: include/drivers/ca/ca_bscan.h (hunk 1 above)
Comment (wavious-wpatty) [reply include/drivers/ca/ca_bscan.h:18]: No DQ will be in the next commit.

File: include/drivers/ca/ca_bscan.h (hunk 2 above)
Comment (wavious-wpatty) [reply include/drivers/ca/ca_bscan.h:33]: No DQS will be in the next commit.

**Review by wavious-jbasista — APPROVED** [review]

**Pull request merged**